group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupMetadataManager.java

```diff
@@ -1211,13 +1192,99 @@ private CoordinatorResult<ConsumerGroupHeartbeatResponseData, Record> consumerGr
         // 1. The member reported its owned partitions;
```

Comment:
do we need this condition because we can only compute a valid assignment if we're given the partitions a member owns?

Reply:
No. This condition is actually slightly incorrect. I will fix it. The idea is to provide a full response when we received a full request from the client. Here, full means that all the fields are set.

Reply:
and if the request is not full then we don't send a full response (i.e. omit the assignment)?

will you be fixing it in a different PR and is there a jira to track it?

Reply:
Did we follow up here as well?

Reply:
I will address this separately: https://issues.apache.org/jira/browse/KAFKA-16367.